src/transaction.js

```diff
@@ -389,11 +389,65 @@ function serializeTx (txObj: TXOBJ, withPrevScriptPubKey: boolean = false): stri
     serializedTx += o.script
   })
 
-  // Only handles forward transfers at the moment
+  // Handles sidechain creation and forward transfers
   if (txObj.version == -4) {
     serializedTx += zbufferutils.numToVarInt(0); // Write every csw
-    serializedTx += zbufferutils.numToVarInt(0); // Write every vsc
-    
+
+    serializedTx += zbufferutils.numToVarInt(txObj.vsc_ccout.length); // Write every vsc
+    for (var i = 0; i < txObj.vsc_ccout.length; i++) {
+
+      const sc_c = txObj.vsc_ccout[i];
+
+      let _b = Buffer.alloc(12);
+      _b.writeUIntLE(sc_c.withdrawalEpochLength, 0, 3);
+      _b.writeIntBE(sc_c.version, 3, 1);
+      zbufferutils.writeUInt64LE(_b, sc_c.value, 4)
+      serializedTx += _b.toString('hex');
+
+      serializedTx += Buffer.from(sc_c.address, 'hex').reverse().toString('hex');
+
+      serializedTx += zbufferutils.getPushDataLength(sc_c.customData)
+      serializedTx += sc_c.customData;
+
+      serializedTx += '01';
+      serializedTx += zbufferutils.getPushDataLength(sc_c.constant)
+      if (sc_c.constant) {
+        serializedTx += sc_c.constant;
+      }
+
+      serializedTx += zbufferutils.getPushDataLength(sc_c.wCertVk)
+      serializedTx += sc_c.wCertVk;
+
+      serializedTx += '01';
+      serializedTx += zbufferutils.getPushDataLength(sc_c.wCeasedVk)
+      if (sc_c.wCeasedVk) {
+        serializedTx += sc_c.wCeasedVk;
+      }
+
+      serializedTx += zbufferutils.getPushDataLength(sc_c.vFieldElementCertificateFieldConfig)
+      if (sc_c.vFieldElementCertificateFieldConfig) {
+        for (let i = 0; i < sc_c.vFieldElementCertificateFieldConfig.length; i++) {
+          serializedTx += sc_c.vFieldElementCertificateFieldConfig[i].toString(16);
```

Comment:
Should this be `buf.writeUInt8(sc_c.vFieldElementCertificateFieldConfig[i])` to match the deserialization?
https://github.com/HorizenOfficial/zencashjs/blob/437c534bc9f766568950951dece4ad7af4e41d12/src/sidechain.js#L152

Reply:
Or if this is correct, let's change it to `.toString('hex')`

Reply:
I'm not a fan of the magic# either, changed it to a named variable for greater clarity.
`serializedTx += sc_c.vFieldElementCertificateFieldConfig[i].toString(hexRadix);`